src/tests/system/tests/test_gdm_smartcard.py

```diff
@@ -0,0 +1,210 @@
+"""
+SSSD Passwordless GDM Smart Card Tests
+
+:requirement: Passwordless GDM
+"""
+
+from __future__ import annotations
+
+import time
+
+import pytest
+from sssd_test_framework.roles.client import Client
+from sssd_test_framework.roles.generic import GenericProvider
+from sssd_test_framework.roles.ipa import IPA
+from sssd_test_framework.roles.ldap import LDAP
+from sssd_test_framework.topology import KnownTopology
+
+
+def client_setup_for_smartcard(client, provider: IPA | LDAP | GenericProvider, pin: str | int | None = None):
+    # Configure SSSD
+    client.authselect.select("sssd", ["with-mkhomedir", "with-smartcard", "with-switchable-auth"])
+    client.sssd.import_domain(provider.domain, provider)
+    client.sssd.config.remove_section("domain/test")
+    client.sssd.default_domain = provider.domain
+    client.sssd.pam["pam_json_services"] = "gdm-switchable-auth"
+    client.sssd.pam["passkey_child_timeout"] = "30"
+
+    if provider.name.lower() != "ldap":
+        client.sssd.pam["pam_cert_auth"] = "True"
+    else:
+        client.sssd.domain["local_auth_policy"] = "enable:passkey"
+
+    client.sssd.start()
+
+
+def enroll_smartcard(client, provider: IPA, username: str, id: str = "01", init: bool = True):
+    cert, key, _ = provider.ca.request(username)
+    cert_content = provider.fs.read(cert)
+    key_content = provider.fs.read(key)
+    client.fs.write(f"/opt/test_ca/{username}_{id}.crt", cert_content)
+    client.fs.write(f"/opt/test_ca/{username}_{id}.key", key_content)
+    if init:
+        client.smartcard.initialize_card()
+    client.smartcard.add_key(f"/opt/test_ca/{username}_{id}.key", key_id=id)
+    client.smartcard.add_cert(f"/opt/test_ca/{username}_{id}.crt", cert_id=id)
+    client.svc.restart("virt_cacard.service")
+
+
+@pytest.mark.topology(KnownTopology.BareIPA)
+@pytest.mark.builtwith(client="gdm")
+def test_gdm__smartcard_login_with_pin(client: Client, ipa: IPA):
+    """
+    :title: Login via GDM using smart card with PIN
+    :setup:
+        1. Configure SSSD for gdm-switchable-auth and pam_cert_auth
+        2. Start SSSD
+        3. Add user to domain
+        4. Enroll smart card in domain for user
+    :steps:
+        1. Login through GDM using smart card with PIN
+    :expectedresults:
+        1. Login successful and user sees home screen
+    :customerscenario: False
+    """
+    testuser = "ipacertuser1"
+    pin = "123456"
+
+    # Configure SSSD and vfido
+    client_setup_for_smartcard(client, ipa, pin=pin)
+
+    # Add IPA User
+    ipa.user(testuser).add()
+
+    # Enroll smartcard with first key/cert issued from IPA
+    enroll_smartcard(client, ipa, testuser, id="01")
+
+    # Authenticate to GDM with smart card
+    client.gdm.click_on("listed?")
+    client.gdm.kb_write(testuser)
+    client.gdm.kb_send("tab")
+    client.gdm.click_on("PIN")
+    client.gdm.kb_write(pin)
+    # time.sleep(5)
+    client.gdm.check_home_screen()
```

Comment:
You should assert every instance of `client.gdm.check_home_screen`

Reply:
added asserts to the check_home_screens().